bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -229,18 +230,27 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
   }
 
   private ResultScanner<Row> streamRows(ReadRowsRequest request) {
-    PooledChannel reservedChannel = channelPool.reserveChannel();
-    final ClientCall<ReadRowsRequest, ReadRowsResponse> readRowsCall =
-        reservedChannel.newCall(BigtableServiceGrpc.METHOD_READ_ROWS, CallOptions.DEFAULT);
+    boolean isGet = request.getTargetCase() == ReadRowsRequest.TargetCase.ROW_KEY;
 
-    // If the scanner is close()d before we're done streaming, we want to cancel the RPC:
-    CancellationToken cancellationToken = new CancellationToken();
-    cancellationToken.addListener(new Runnable() {
-      @Override
-      public void run() {
-        readRowsCall.cancel();
-      }
-    }, executorService);
+    // Reserve a channel if this is a streaming call, since gRPC does
+
+    PooledChannel reservedChannel =  isGet ? null : channelPool.reserveChannel();
```

Comment:
We want to use a channel from the pool without reserving it if this is a Get.  Calling  channelPool.reserveChannel() doesn't seem to make sense to me.